packages/frontend2/src/app/_components/chart/costs/use-cost-chart-render-params.ts

```diff
@@ -0,0 +1,186 @@
+import { type Milestone } from '@l2beat/config'
+import { assertUnreachable } from '@l2beat/shared-pure'
+import { useCallback, useMemo } from 'react'
+import {
+  type CostsChartResponse,
+  type CostsUnit,
+} from '~/server/features/scaling/costs/types'
+import { formatCurrency } from '~/utils/format'
+import { formatNumber } from '~/utils/format-number'
+import { type SeriesStyle } from '../core/styles'
+import { mapMilestones } from '../core/utils/map-milestones'
+import { type CostsChartPointData } from './costs-chart-hover'
+
+const DENCUN_UPGRADE_TIMESTAMP = 1710288000
+
+interface Params {
+  milestones: Milestone[]
+  unit: CostsUnit
+  chart: CostsChartResponse | undefined
+}
+
+export function useCostChartRenderParams({ milestones, unit, chart }: Params) {
+  const mappedMilestones = useMemo(
+    () => mapMilestones(milestones),
+    [milestones],
+  )
+
+  const formatYAxisLabel = useCallback(
+    (value: number) =>
+      unit === 'gas'
+        ? formatNumber(value)
+        : formatCurrency(value, unit, { showLessThanMinimum: false }),
+    [unit],
+  )
+
+  const columns = useMemo(
+    () =>
+      chart?.data.map((dataPoint) => {
+        const [timestamp] = dataPoint
+
+        return {
+          values: getValues(dataPoint, unit),
+          data: getData(dataPoint, unit),
+          milestone: mappedMilestones[timestamp],
+        }
+      }) ?? [],
+    [chart?.data, mappedMilestones, unit],
+  )
+
+  const chartRange: [number, number] = useMemo(
+    () => [
+      chart?.data[0]?.[0] ?? 0,
+      chart?.data[chart.data.length - 1]?.[0] ?? 1,
+    ],
+    [chart?.data],
+  )
+
+  const valuesStyle: SeriesStyle[] = useMemo(
+    () => [
+      {
+        line: 'blue',
+        fill: 'blue',
+        point: 'circle',
+      },
+      {
+        line: 'light-yellow',
+        fill: 'light-yellow',
+      },
+      {
+        line: 'pink',
+        fill: 'pink',
+      },
+      {
+        line: 'purple',
+        fill: 'purple',
+      },
+    ],
+    [],
+  )
+
+  return {
+    formatYAxisLabel,
+    columns,
+    chartRange,
+    valuesStyle,
+  }
+}
+
+function getValues(
+  dataPoint: CostsChartResponse['data'][number],
+  unit: CostsUnit,
+) {
+  const [
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
```

Comment:
Let's just modify the eslint rule to allow using values prefixed by `_` in these kind of cases.

Reply:
good call